src/commands/ext-update.ts

```diff
@@ -60,148 +68,219 @@ export default new Command("ext:update <extensionInstanceId> [localDirectoryOrUr
+            SourceOrigin.URL,
+            SourceOrigin.PUBLISHED_EXTENSION,
+            SourceOrigin.PUBLISHED_EXTENSION_VERSION,
+          ].indexOf(newSourceOrigin) > -1
+        ) {
+          validUpdate = true;
         }
-        utils.logLabeledBullet(logPrefix, `${clc.bold(msg1)}\n`);
-        let updateWarning: resolveSource.UpdateWarning;
-        let updatingFromOfficial = false;
-        try {
-          const registryEntry = await resolveSource.resolveRegistryEntry(currentSpec.name);
-          updatingFromOfficial = resolveSource.isOfficialSource(registryEntry, existingSource);
-        } catch {
-          // If registry entry does not exist, assume local directory or URL extension source.
+      } else if (
+        existingSourceOrigin === SourceOrigin.LOCAL ||
+        existingSourceOrigin === SourceOrigin.URL
+      ) {
+        if ([SourceOrigin.LOCAL, SourceOrigin.URL].indexOf(newSourceOrigin) > -1) {
+          validUpdate = true;
         }
+      }
+      if (!validUpdate) {
+        throw new FirebaseError(
+          `Cannot update from a(n) ${existingSourceOrigin} to a(n) ${newSourceOrigin}. Please provide a new source that is a(n) ${existingSourceOrigin} and try again.`
+        );
+      }
 
-        if (updatingFromOfficial) {
-          updateWarning = {
-            from: "",
-            description: `${msg2}\n\n${msg3}`,
-          };
-        } else {
-          updateWarning = {
-            from: "",
-            description: `${msg2}`,
-          };
-        }
+      const isPublished =
+        [
+          SourceOrigin.OFFICIAL_EXTENSION,
+          SourceOrigin.OFFICIAL_EXTENSION_VERSION,
+          SourceOrigin.PUBLISHED_EXTENSION,
+          SourceOrigin.PUBLISHED_EXTENSION_VERSION,
+        ].indexOf(newSourceOrigin) > -1;
+      displayExtInfo(instanceId, existingSpec, isPublished);
 
-        await confirmUpdateWarning(updateWarning);
-      } else {
-        // Updating to a version from an official source
-        let registryEntry;
-        try {
-          registryEntry = await resolveSource.resolveRegistryEntry(currentSpec.name);
-        } catch (err) {
-          // If registry entry does not exist, assume local directory or URL extension source.
-          throw new FirebaseError(
-            `Unable to update this instance without a local or URL source. To update this instance, run "firebase ext:update ${instanceId} <localDirectoryOrUrl>".`
+      // TODO: remove "falls through" once producer and registry experience are released
+      switch (newSourceOrigin) {
+        case SourceOrigin.LOCAL: {
+          if (previews.extdev) {
+            newSourceName = await updateFromLocalSource(
+              projectId,
+              instanceId,
+              updateSource,
+              existingSpec,
+              existingSource
+            );
+            break;
+          }
+          // falls through
+        }
+        // eslint-disable-next-line no-fallthrough
```

Comment:
I _think_ this can be omitted since you say `// falls through` above?

If that's not the case, it's because you're including `{}` after the `case` statement which is unnecessary - they should probably be removed

Reply:
I removed the { } - good call on that! Now I only need to disable eslint once using // eslint-disable-next-line no-fallthrough (rather than copy/paste it several times).

For the // falls through comment, though, I still think I need to disable eslint at least once, because the linter says that you're NOT allowed to write case statements that fall through to the default after falling into a case statement. We want exactly that behavior in this case.